lib/scholar/linear/polynomial_regression.ex

```diff
@@ -0,0 +1,226 @@
+      >
+      iex> model.intercept
+      #Nx.Tensor<
+        f32
+        5.896470069885254
+      >
+      iex> model.degree
+      1
+
+      iex> x = Nx.tensor([[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]])
+      iex> y = Nx.tensor([4.0, 3.0, -1.0])
+      iex> model = Scholar.Linear.PolynomialRegression.fit(x, y, degree: 2)
+      iex> model.coefficients
+      #Nx.Tensor<
+        f32[5]
+        [-0.021396497264504433, -0.004854594357311726, -0.0884987860918045, -0.062211357057094574, -0.04369127005338669]
+      >
+      iex> model.intercept
+      #Nx.Tensor<
+        f32
+        4.418517112731934
+      >
+      iex> model.degree
+      2
+  """
+  deftransform fit(a, b, opts \\ []) do
+    opts = NimbleOptions.validate!(opts, @opts_schema)
+    a_transform = transform(a, fit_intercept?: false, degree: opts[:degree])
+
+    linear_reg =
+      Scholar.Linear.LinearRegression.fit(a_transform, b, Keyword.take(opts, [:fit_intercept?]))
+
+    %__MODULE__{
+      coefficients: linear_reg.coefficients,
+      intercept: linear_reg.intercept,
+      degree: opts[:degree]
+    }
+  end
+
+  @doc """
+  Makes predictions with the given `model` on input `x`.
+
+  ## Examples
+
+      iex> x = Nx.tensor([[1.0, 2.0], [3.0, 2.0], [4.0, 7.0]])
+      iex> y = Nx.tensor([4.0, 3.0, -1.0])
+      iex> model = Scholar.Linear.PolynomialRegression.fit(x, y, degree: 2)
+      iex> Scholar.Linear.PolynomialRegression.predict(model, Nx.tensor([[2.0, 1.0]]))
+      #Nx.Tensor<
+        f32[1]
+        [3.8487603664398193]
+      >
+  """
+  deftransform predict(model, x) do
+    Scholar.Linear.LinearRegression.predict(
+      %Scholar.Linear.LinearRegression{
+        coefficients: model.coefficients |> Nx.flatten(),
+        intercept: model.intercept
+      },
+      transform_n(x, degree: model.degree, fit_intercept?: false)
+    )
+  end
+
+  @doc """
+  Computes the feature matrix for polynomial regression.
+
+  #{NimbleOptions.docs(@transform_opts_schema)}
+
+  ## Examples
+
+      iex> x = Nx.tensor([[2]])
+      iex> Scholar.Linear.PolynomialRegression.transform(x, degree: 0)
+      ** (NimbleOptions.ValidationError) invalid value for :degree option: expected positive integer, got: 0
+
+      iex> x = Nx.tensor([[2]])
+      iex> Scholar.Linear.PolynomialRegression.transform(x, degree: 5, fit_intercept?: false)
+      #Nx.Tensor<
+        s64[1][5]
+        [
+          [2, 4, 8, 16, 32]
+        ]
+      >
+
+      iex> x = Nx.tensor([[2, 3]])
+      iex> Scholar.Linear.PolynomialRegression.transform(x)
+      #Nx.Tensor<
+        s64[1][6]
+        [
+          [1, 2, 3, 4, 6, 9]
+        ]
+      >
+
+      iex> x = Nx.iota({3, 2})
+      iex> Scholar.Linear.PolynomialRegression.transform(x, fit_intercept?: false)
+      #Nx.Tensor<
+        s64[3][5]
+        [
+          [0, 1, 0, 0, 1],
+          [2, 3, 4, 6, 9],
+          [4, 5, 16, 20, 25]
+        ]
+      >
+  """
+  deftransform transform(x, opts \\ []) do
```

Comment:
Why is it called `transform` instead of `predict` like in linear regression?

Reply:
`predict` is still there. `transform` only computes the feature matrix.

Reply:
I see, thanks. Why is it called `transform`? Is this a convention that comes from scikit-learn? Or should we call it `feature_matrix` or similar? Thoughts? /cc @polvalente

Reply:
yup, convention from sklearn (https://towardsdatascience.com/what-and-why-behind-fit-transform-vs-transform-in-scikit-learn-78f915cf96fe)